src/storage/ring_buffer.rs

```diff
@@ -101,6 +119,67 @@ impl<'a, T: 'a> RingBuffer<'a, T> {
             Err(error) => Err(error)
         }
     }
+
+    /// Get capacity of the underlying storage.
+    pub fn capacity(&self) -> usize {
+        self.storage.len()
+    }
+
+    /// Remove the first element equal to `value` from the buffer,
+    /// reutrn `Err(Error::Exhausted)` if no such element was found.
+    pub fn remove(&mut self, value: &T) -> Result<()>
```

Comment:
This seems to go completely against the spirit of a ring buffer. Why do you need this?

Reply:
I was using `RingBuffer` as a queue of dirty sockets and I needed to delete sockets from the queue once they're removed from the main container. I agree its usefulness isn't evident, but it could be handy to have.

Reply:
Ah. But as I've explained in [this comment](https://github.com/m-labs/smoltcp/issues/19#issuecomment-325299849) a queue won't work here, you'll need a BTreeMap as well, see my latest commit for more context as to why.

Reply:
Ok then.